### PR TITLE
Protected roles tweaks and refactor

### PR DESCRIPTION
**Pull request by ShiftyRail**
[system] [tweak] [role]

Rather than a single, universal probabilistic roll, rulesets will now look for a prob chance defined in the role.
Rulesets use a role type rather than an id to reduce the boilerplate.
In the ~~near~~ future, role ids should be avoided altogether and typepaths should be used, notably for the protected roles list itself.

Tested ; the probabilistic roll does take place, normal jobs still get their antag roll.

As an application, the probability for an officer to roll traitor successfully is 50%, but to roll vamp is 20%.
Meant as an alternative to #21034.
Can be further tweaked if needed.

:cl:
- tweak: NanoTrasen has changed their screening process for their security and command forces, resulting in less vampires being hired.

### CODE REVIEW

File: __DEFINES/setup.dm [hunk 1]
```diff
@@ -871,6 +871,17 @@ SEE_PIXELS	256
 #define ROLE_VOXRAIDER  	"vox raider"
 #define ROLE_WIZARD     	"wizard"
 
+// Role typepaths.
+
+#define ROLE_CULTIST_TYPE /datum/role/cultist
```
Comment (DamianX) [inline __DEFINES/setup.dm]: Why not use the type directly? what's the be gained from this?

Comment (ShiftyRail) [reply __DEFINES/setup.dm]: Not much, in retrospect.

**Comment by Intigracy** [comment]
Even though it's removed from roundstart, make ling have the rare probability.

**Comment by ShiftyRail** [comment]
Understood.

**Pull request merged**